manifests/fpm.pp

```diff
@@ -48,7 +48,7 @@
   anchor { '::php::fpm::end': }
 
   $real_pools = hiera_hash('php::fpm::pools',  $pools)
-  create_resources(::php::fpm::pool, $real_pools)
+  create_resources('php::fpm::pool', $real_pools)
```

Comment:
Does `'::php::fpm::pool'` also solve this problem for you? It's generally better and faster to use the fully qualified form to refer to classes and resources.